dev-lang/ghc/ghc-9.8.4-r1.ebuild

```
@@ -255,7 +261,10 @@ bump_lib() {
 
 	einfo "Bumping ${pn} up to ${pv}"
 
-	mv "${dir}"/"${pn}" "${WORKDIR}"/"${pn}".old || die
+	if [-d "${dir}"/"${pn}"]
+	then
+		mv "${dir}"/"${pn}" "${WORKDIR}"/"${pn}".old || die
+	fi
```

I don't understand the purpose of this change. Wouldn't we want `die` to be called in the event the directory doesn't exist, rather than silently skipping it?

---

Well, that's the issue: ghc-9.6.7 won't compile without file-io, but file-io is not in the tarball, so we need to add it. It's more an ADD_LIB than a BUMP_LIB thing, I realise. Want me to write a separate function?

---

> Everything builds (well, except for dev-haskell/hip, but it was already not building)

It's not formatted correctly, so I think it skipped all the updates. You may have to redo the build-test.

```
 * Bumping directory up to 1.3.9.0
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
 * Bumping file-io up to 0.1.5
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
 * Bumping filepath up to 1.5.4.0
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
```

(Since it fails with "command not found" `mv` never gets run.)

---

Nvm, I didn't understand the full context

```ebuild
    if [[ -d "${dir}"/"${pn}" ]]
    then
        mv "${dir}"/"${pn}" "${WORKDIR}"/"${pn}".old || die
    fi
    mv "${WORKDIR}"/"${p}" "${dir}"/"${pn}" || die
```

---

In any case it's misformatted

```
 * Bumping directory up to 1.3.9.0
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
 * Bumping file-io up to 0.1.5
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
 * Bumping filepath up to 1.5.4.0
/var/tmp/portage/dev-lang/ghc-9.8.4-r1/temp/environment: line 1716: [-d: command not found
```

```suggestion
	if [[ -d "${dir}"/"${pn}" ]]
	then
		mv "${dir}"/"${pn}" "${WORKDIR}"/"${pn}".old || die
	fi
```

---

If the old directory didn't get moved, then this would move the new version in as a subdirectory of the old version:

```ebuild
mv "${WORKDIR}"/"${p}" "${dir}"/"${pn}" || die
```

---

You should double-check they actually got bumped

```bash
for p in file-io directory filepath; do
	ghc-pkg field $p version
done
```